examples/asyncio/wsgi-server.py

```diff
@@ -688,41 +673,41 @@ def _build_environ_dict(headers, stream):
     version you'd want to fix it.
     """
     header_dict = dict(headers)
-    path = header_dict.pop(u':path')
+    path = header_dict.pop(':path')
     try:
-        path, query = path.split(u'?', 1)
+        path, query = path.split('?', 1)
     except ValueError:
-        query = u""
-    server_name = header_dict.pop(u':authority')
+        query = ""
+    server_name = header_dict.pop(':authority')
     try:
-        server_name, port = server_name.split(u':', 1)
-    except ValueError as e:
+        server_name, port = server_name.split(':', 1)
+    except ValueError:
         port = "8443"
 
     environ = {
-        u'REQUEST_METHOD': header_dict.pop(u':method'),
-        u'SCRIPT_NAME': u'',
-        u'PATH_INFO': path,
-        u'QUERY_STRING': query,
-        u'SERVER_NAME': server_name,
-        u'SERVER_PORT': port,
-        u'SERVER_PROTOCOL': u'HTTP/2',
-        u'HTTPS': u"on",
-        u'SSL_PROTOCOL': u'TLSv1.2',
-        u'wsgi.version': (1, 0),
-        u'wsgi.url_scheme': header_dict.pop(u':scheme'),
-        u'wsgi.input': stream,
-        u'wsgi.errors': sys.stderr,
-        u'wsgi.multithread': True,
-        u'wsgi.multiprocess': False,
-        u'wsgi.run_once': False,
+        'REQUEST_METHOD': header_dict.pop(':method'),
+        'SCRIPT_NAME': '',
+        'PATH_INFO': path,
+        'QUERY_STRING': query,
+        'SERVER_NAME': server_name,
+        'SERVER_PORT': port,
+        'SERVER_PROTOCOL': 'HTTP/2',
+        'HTTPS': "on",
+        'SSL_PROTOCOL': 'TLSv1.2',
+        'wsgi.version': (1, 0),
+        'wsgi.url_scheme': header_dict.pop(':scheme'),
+        'wsgi.input': stream,
+        'wsgi.errors': sys.stderr,
+        'wsgi.multithread': True,
+        'wsgi.multiprocess': False,
+        'wsgi.run_once': False,
     }
-    if u'content-type' in header_dict:
-        environ[u'CONTENT_TYPE'] = header_dict[u'content-type']
-    if u'content-length' in header_dict:
-        environ[u'CONTENT_LENGTH'] = header_dict[u'content-length']
+    if 'content-type' in header_dict:
+        environ['CONTENT_TYPE'] = header_dict.pop('content-type')
+    if 'content-length' in header_dict:
+        environ['CONTENT_LENGTH'] = header_dict.pop('content-length')
```

Comment:
Any specific reason for using `pop` instead of `get` here?
I feel like using `get` would be more intuitive, unless you intended the side effect?

Reply:
https://wsgi.readthedocs.io/en/latest/definitions.html#envvar-CONTENT_TYPE
yes, both `CONTENT_TYPE` and `CONTENT_LENGTH` headers should stay "as-is" without prefixing, otherwise we will pass them to the environ with `HTTP_`, that's why we remove them (see same for the request method above)

Reply:
Thanks - now it makes sense!